.travis.yml

```diff
@@ -33,16 +33,19 @@ matrix:
     include:
         - os: linux
           compiler: clang-3.6
-          env: CONFIG_OPTS="-fsanitize=address no-shared"
+          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 -DPEDANTIC"
         - os: linux
           compiler: clang-3.6
-          env: CONFIG_OPTS="no-shared no-asm -fno-sanitize-recover -fsanitize=address -fsanitize=undefined enable-rc5 enable-md2 -fno-sanitize=alignment"
+          env: CONFIG_OPTS="no-asm enable-ubsan enable-rc5 enable-md2 -fno-sanitize=alignment"
+        - os: linux
+          compiler: clang-3.6
+          env: CONFIG_OPTS="enable-asan no-shared"
         - os: linux
           compiler: gcc-5
-          env: CONFIG_OPTS="no-shared -fsanitize=address"
+          env: CONFIG_OPTS="no-asm enable-asan enable-rc5 enable-md2"
         - os: linux
           compiler: gcc-5
-          env: CONFIG_OPTS="no-shared no-asm -fno-sanitize-recover -DPEDANTIC -fsanitize=address -fsanitize=undefined enable-rc5 enable-md2"
+          env: CONFIG_OPTS="no-asm enable-asan enable-rc5 enable-md2 -DPEDANTIC"
```

Comment:
With no-asm, either -fno-sanitize=alignment or -DPEDANTIC is needed:  without DPEDANTIC, some platform-specific crypto code upsets the alignment sanitizer. So we have two different ubsan runs, one for each. They end up exercising slightly different code paths.